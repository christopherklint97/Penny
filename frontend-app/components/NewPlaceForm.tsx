import React from 'react'
import { useForm } from 'react-hook-form'

interface NewPlaceFormProps {
  onSubmit: (data: any) => void
}

export default function NewPlaceForm({ onSubmit }: NewPlaceFormProps) {
  const { register, handleSubmit, errors } = useForm()

  return (
    /* "handleSubmit" will validate your inputs before invoking "onSubmit" */
    <form className="text-center" onSubmit={handleSubmit(onSubmit)}>
      <input
        className="px-2 py-1 shadow-md focus:ring-2 focus:outline-none rounded-md mr-3 mb-2 w-3/4"
        name="place"
        ref={register({ required: true })}
        placeholder="Start typing for place suggestions..."
      />
      {/* errors will return when field validation fails  */}
      {errors.city && (
        <p className="text-red-900 text-sm">This field is required</p>
      )}

      <input
        className="px-2 py-1 bg-green-200 rounded-md shadow-md hover:bg-green-300 mb-4 cursor-pointer"
        type="submit"
        value="Add"
      />
    </form>
  )
}
