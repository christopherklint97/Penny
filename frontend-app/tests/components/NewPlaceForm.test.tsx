import React from 'react'
import { render } from '../testUtils'
import NewPlaceForm from '../../components/NewPlaceForm'

describe('NewPlaceForm component', () => {
  it('renders without crashing', function () {
    render(<NewPlaceForm onSubmit={jest.fn()} />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<NewPlaceForm onSubmit={jest.fn()} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
