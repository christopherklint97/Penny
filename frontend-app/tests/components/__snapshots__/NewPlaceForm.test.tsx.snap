// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPlaceForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="text-center"
  >
    <input
      class="px-2 py-1 shadow-md focus:ring-2 focus:outline-none rounded-md mr-3 mb-2 w-3/4"
      name="place"
      placeholder="Start typing for place suggestions..."
    />
    <input
      class="px-2 py-1 bg-blue-300 hover:bg-blue-400 text-gray-900 rounded-md shadow-md mb-4 cursor-pointer"
      type="submit"
      value="Add"
    />
  </form>
</DocumentFragment>
`;
